### ISSUE TITLE
[BUG] Metamask connector is not found in the first load

### ISSUE THREAD

**Issue by caotrongtin99** [wevm/wagmi#3679]
### Describe the bug

Hi team, I am developing a web3 dapp using wagmi and nextjs, i just faced this problem recently. I installed the Coinbase extension and Metamask extension on my browser. When I access to the dapp, on the first load, then I click Metamask icon to connect. The result is that the metamask popup was not opened. The error on console tab is:
```
ConnectorNotFoundError: Connector not found
    at MetaMaskConnector.connect (webpack-internal:///./node_modules/@wagmi/connectors/dist/metaMask.js:87:15)
    at async connect (webpack-internal:///./node_modules/@wagmi/core/dist/chunk-TSH6VVF4.js:720:18)
```
But when i try to reload the page, it can connect to metamask wallet.

### Link to Minimal Reproducible Example
(https://codesandbox.io/p/github/caotrongtin99/metamask-reproduce/main)

### Steps To Reproduce

1. Installed both Metamask and Coinbase wallet extensions on your browser
2. Open the codesandbox (https://codesandbox.io/p/github/caotrongtin99/metamask-reproduce/main)
3. Running the code
4. In the first load, when u click Metamask, it will show the error (Connector not found)
5. Try to refresh the browser in CodeSandbox, u can connect to Metamask wallet.

### Wagmi Version

1.4.7

### Viem Version

1.19.4

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

**Comment by caotrongtin99** [comment]
hi @tmm , i have reproduced steps, could u help to check? Thanks

**Comment by glitch-txs** [comment]
not able to repro, your MRE works for me, but this might be due to delay injection which is fixed in Wagmi v2

**Comment by caotrongtin99** [comment]
> not able to repro, your MRE works for me, but this might be due to delay injection which is fixed in Wagmi v2

Hi @glitch-txs, have u installed Metamask and Coinbase yet?

**Comment by caotrongtin99** [comment]
@glitch-txs  I have attached a video regarding bug repro (https://vimeo.com/manage/videos/921753012). Could u try by following the steps:
1. Install Metamask and Coinbase
2. Exit the Chrome browser
3. Open the codesandbox
4. Click Metamask, the error message will be shown
5. Refresh page in the codesandbox, we can connect to metamask.
Besides, I have tried to upgrade wagmi to v2 and viem to v2, it didn't work.
